core/src/main/scala/akka/kafka/internal/ConsumerStage.scala

```diff
@@ -435,9 +436,11 @@ private[kafka] abstract class ConsumerStageLogic[K, V, Out](
   // impl of Consumer.Control that is called from outside via materialized value
   override def isShutdown: Future[Done] = shutdownPromise.future
 
-  override def stop(): Unit = {
+  override def stop(): Future[Done] = {
     log.debug("Stop producing messages")
-    stopCallback.invoke(())
+    val result = Promise[Done]
```

Comment:
There is a problem with this. If the stage is already shutdown, this promise will never be completed. Better to implement it as is done for `shutdown`, i.e. with a `val stopPromise` in the stage logic.